server/handlers/converters.go

```diff
@@ -67,16 +67,32 @@ func (ah *ActivityHandler) handleMentions(msg *msteams.Message) string {
 	for _, mention := range msg.Mentions {
 		mmMention := ""
 		if mention.UserID != "" {
+			var mmUser *model.User
+			var getErr *model.AppError
 			mmUserID, err := ah.plugin.GetStore().TeamsToMattermostUserID(mention.UserID)
 			if err != nil {
-				ah.plugin.GetAPI().LogDebug("Unable to get mm UserID", "Error", err.Error())
-				continue
-			}
-
-			mmUser, getErr := ah.plugin.GetAPI().GetUser(mmUserID)
-			if getErr != nil {
-				ah.plugin.GetAPI().LogDebug("Unable to get mm user details", "Error", getErr.Error())
-				continue
+				msUser, mErr := ah.plugin.GetClientForApp().GetUser(mention.UserID)
```

Comment:
If we are going to identify the user by the email, we should have a way to store that email information directly in the local store and use that whenever is needed. If not this is going to trigger a lot of API calls that are not needed. In the sample places where we store the MSTeams User ID we should store the MSTeams Email, and the add another method to get the User Email based on the mattermost userID, for example.

Reply:
@jespino Storing the email information in the store will not help in reducing the API calls as these are the cases when the user details are not present in the store itself. So even if we store the email information in the store, we cannot get user details from the store.